http/http-advanced/src/test/java/io/quarkus/ts/openshift/http/clients/CommonClientService.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.ts.openshift.http.clients;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+public abstract class CommonClientService {
+
+    protected String uri;
+    protected String password;
+    protected boolean hostVerified;
+    protected String keyStorePath;
+
+    public CommonClientService withPassword(String password) {
+        this.password = password;
+        return this;
+    }
+
+    public CommonClientService withHostVerified(Boolean verified) {
+        this.hostVerified = verified;
+        return this;
+    }
+
+    public CommonClientService withKeyStorePath(String keyStorePath) {
+        this.keyStorePath = keyStorePath;
+        return this;
+    }
+
+    public abstract <T> T build() throws Exception;
```

Comment:
This is returning a warning because a possible cast exception. It should use generics at class level: CommonClientService<T> and change the implementations accordingly.